cocos/core/data/utils/compact-value-type-array.ts

```diff
@@ -139,11 +142,11 @@ export class CompactValueTypeArray {
     }
 }
 
-function getElementTraits (elementType: ElementType) {
+function getElementTraits (elementType: ElementType): CompactTraits {
     return BuiltinElementTypeTraits[elementType];
 }
 
-function getStorageConstructor (unit: StorageUnit) {
+function getStorageConstructor (unit: StorageUnit): Float32ArrayConstructor | Uint16ArrayConstructor | Uint32ArrayConstructor | Uint8ArrayConstructor | Int8ArrayConstructor | Int16ArrayConstructor | Int32ArrayConstructor | Float64ArrayConstructor {
```

Comment:
Is there  ArrayContructor that covers all these types.

Reply:
we do have one called `TypedArrayConstructor`

Reply:
updated https://github.com/cocos/cocos-engine/pull/15314/commits/8843bfe54cd3f8e6f84432948716c8d130ecb233